core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1239,10 +1242,30 @@ private void unParseRegexpLike(SqlWriter writer, SqlCall call, int leftPrec, int
     writer.endFunCall(ifFrame);
   }
 
-  private void unparseShiftLeft(SqlWriter writer, SqlCall call) {
+  private void unparseShiftLeftAndShiftRight(SqlWriter writer, SqlCall call, boolean isShiftLeft) {
```

Comment:
can we refactor this code and extract some code to function to make it more clear.
I have create 3 methods and separate out the code to make it more clear.
1) isBasicCallWithNegativePrefix
2) getPositiveOperand
3) getShiftOperator
eg : 


private void unparseShiftLeftAndShiftRight(SqlWriter writer, SqlCall call, boolean isShiftLeft) {
    call.operand(0).unparse(writer, 0, 0);
    SqlNode secondOperand = call.operand(1);

    if (isBasicCallWithNegativePrefix(secondOperand)) {
        SqlNode positiveOperand = getPositiveOperand(secondOperand);
        writer.print(getShiftOperator(isShiftLeft));
        writer.print(" ");
        positiveOperand.unparse(writer, 0, 0);
    } else {
        writer.print(getShiftOperator(isShiftLeft));
        writer.print(" ");
        secondOperand.unparse(writer, 0, 0);
    }
}


private boolean isBasicCallWithNegativePrefix(SqlNode secondOperand) {
    return secondOperand instanceof SqlBasicCall &&
            ((SqlBasicCall) secondOperand).getOperator().getKind() == SqlKind.MINUS_PREFIX;
}

private SqlNode getPositiveOperand(SqlNode secondOperand) {
    return ((SqlNode[]) ((SqlBasicCall) secondOperand).operands)[0];
}

private String getShiftOperator(boolean isShiftLeft) {
    return isShiftLeft ? SHIFTLEFT : SHIFTRIGHT;
}

Reply:
also check if ((SqlNode[]) ((SqlBasicCall) secondOperand).operands)[0]; condition can be replaced by ((SqlBasicCall) firstOperand).operand(0)

Reply:
Done

Reply:
changes looks good.